src/probnum/filtsmooth/gaussfiltsmooth/kalmanposterior.py

```diff
@@ -141,11 +114,88 @@ def sample(self, locations=None, size=()):
             [self.sample(locations=locations, size=size[1:]) for _ in range(size[0])]
         )
 
+    @abc.abstractmethod
     def _single_sample_path(self, locations, random_vars):
+        raise NotImplementedError
+
+    # If these two functions can be combined reliably into one
+    # function, please feel free to make the changes.
```

Comment:
the "<" and "<=" would lead to some awkward corner cases IMO. Do you agree?

Reply:
I do agree that merging them is not a good idea. But as commented above, I think we don't actually need "previous or same". The code I had before was just a weird implementation of "find the index where t_i == t" (it's probably more efficient than a "where" because the locations are sorted and it was really just the simplest way of implementing this that came to my mind).